handlers/public.go

```diff
@@ -0,0 +1,208 @@
+	ID       string `bson:"id"`
+	Surname  string `bson:"surname"`
+}
+
+// Links is a set of URLs related to the resource, including self
+type Links struct {
+	Journey  string `json:"journey"`
+	Resource string `json:"resource"`
+	Self     string `json:"self"`
+}
+
+//Data is a representation of the top level data retrieved from the Transaction API
+type Data struct {
+	CompanyName string            `json:"company_name"`
+	Filings     map[string]Filing `json:"filings"`
+}
+
+//Filing is a representation of the Filing subsection of data retrieved from the Transaction API
+type Filing struct {
+	Description string `json:"description"`
+}
+
+// Register defines the route mappings
+func Register(r *pat.Router) {
+	r.Get("/healthcheck", getHealthCheck).Name("get-healthcheck")
+	r.Post("/payments", createPaymentSession).Name("create-payment")
+}
+
+func getHealthCheck(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+}
+
+var client http.Client
+
+func createPaymentSession(w http.ResponseWriter, req *http.Request) {
+	if req.Body == nil {
+		log.ErrorR(req, fmt.Errorf("request body empty"))
+		w.WriteHeader(http.StatusBadRequest) // 400
+		return
+	}
+
+	requestDecoder := json.NewDecoder(req.Body)
+	var b createPaymentResource
+	err := requestDecoder.Decode(&b)
+	if err != nil {
+		log.ErrorR(req, fmt.Errorf("request body invalid"))
+		w.WriteHeader(http.StatusBadRequest) // 400
+		return
+	}
+
+	resource := b.Resource
+
+	cfg, err := config.Get()
+	if err != nil {
+		log.ErrorR(req, fmt.Errorf("error getting config: [%s]", err))
+		w.WriteHeader(http.StatusInternalServerError) // 500
+		return
+	}
+	url, err := url.Parse(resource)
+	if err != nil {
+		log.ErrorR(req, fmt.Errorf("error parsing resource: [%s]", err))
+		w.WriteHeader(http.StatusBadRequest) // 400
+		return
+	}
+
+	whitelist := strings.Split(cfg.DomainWhitelist, ",")
+	matched := false
+	for _, domain := range whitelist {
+		if url.Host == domain {
+			matched = true
+			break
+		}
+	}
+	if !matched {
+		log.ErrorR(req, fmt.Errorf("invalid resource domain: %s", url.Host))
+		w.WriteHeader(http.StatusBadRequest) // 400
+		return
+	}
+
+	resourceReq, err := http.NewRequest("GET", resource, nil)
+	if err != nil {
+		log.ErrorR(resourceReq, fmt.Errorf("failed to create Resource Request: [%s]", err))
+		w.WriteHeader(http.StatusInternalServerError) // 500
+		return
+	}
+	resp, err := client.Do(resourceReq)
+	if err != nil {
+		log.ErrorR(resourceReq, fmt.Errorf("error getting Cost Resource: [%s]", err))
+		w.WriteHeader(http.StatusInternalServerError) // 500
+		return
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.ErrorR(resourceReq, fmt.Errorf("error reading Cost Resource: [%s]", err))
+		w.WriteHeader(http.StatusInternalServerError) // 500
+		return
+	}
+
+	// TODO save cost resource and ensure all mandatory fields are present:
+
+	paymentResource := &PaymentResource{}
+	err = json.Unmarshal(body, paymentResource)
```

Comment:
same as above could just use `json.Unmarshal(body, paymentResource)` in if statement instead of assigning locally